### PR TITLE
refactor: Add tests and support pylint

### PR DESCRIPTION
**Pull request by moeez96**
# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/extensions/iap/api/v1/google_validator.py [hunk 1]
```diff
@@ -11,16 +11,23 @@ def validate(self, receipt, configuration):
...
+                is_subscription=False
+            )
             result = {
                 'raw_response': response.raw_response,
                 'is_canceled': response.is_canceled,
                 'is_expired': response.is_expired
             }
         except errors.GoogleError as exc:
-            logger.error('Purchase validation failed {}'.format(exc))
+            logger.error('Purchase validation failed %s', exc)
```
Comment (jawad-khan) [inline ecommerce/extensions/iap/api/v1/google_validator.py:26]: why did you choose % against .format?

File: ecommerce/extensions/iap/api/v1/google_validator.py [hunk 2]
```diff
@@ -11,16 +11,23 @@ def validate(self, receipt, configuration):
...
         purchase_token = receipt['purchaseToken']
         product_sku = receipt['productId']
-        verifier = GooglePlayVerifier(configuration.get('google_bundle_id'),configuration.get('google_service_account_key_file'),)
+        verifier = GooglePlayVerifier(
+            configuration.get('google_bundle_id'),
+            configuration.get('google_service_account_key_file'),
+        )
         try:
-            response = verifier.verify_with_result(purchase_token, product_sku, is_subscription=False)
+            response = verifier.verify_with_result(
```
Comment (jawad-khan) [inline ecommerce/extensions/iap/api/v1/google_validator.py]: try not t break every line which can be written in single line.

File: ecommerce/extensions/iap/processors/base_iap.py [hunk 3]
```diff
@@ -111,28 +112,26 @@ def handle_processor_response(self, response, basket=None):
                 )
                 raise GatewayError(validation_response)
 
-        transaction_id = response.get('transactionId')
-        if not transaction_id:
-            transaction_id = self._get_attribute_from_receipt(validation_response, 'transaction_id')
+        transaction_id = response.get('transactionId', self._get_attribute_from_receipt(
```
Comment (jawad-khan) [inline ecommerce/extensions/iap/processors/base_iap.py]: Handle transaction id according to our discussion.

File: ecommerce/extensions/iap/api/v1/google_validator.py (hunk 1 above)
Comment (moeez96) [reply ecommerce/extensions/iap/api/v1/google_validator.py:26]: Pylint prefers string interpolation in logger to be done via %s
Reference: https://pylint.pycqa.org/en/latest/user_guide/messages/warning/logging-format-interpolation.html

File: ecommerce/extensions/iap/api/v1/google_validator.py (hunk 2 above)
Comment (moeez96) [reply ecommerce/extensions/iap/api/v1/google_validator.py]: This is done for code to look more aesthetic. We can discuss further on num. of lines vs aesthetics and make changes accordingly.

File: ecommerce/extensions/iap/processors/base_iap.py (hunk 3 above)
Comment (moeez96) [reply ecommerce/extensions/iap/processors/base_iap.py]: Done

File: ecommerce/extensions/iap/processors/base_iap.py [hunk 4]
```diff
@@ -173,8 +172,17 @@ def issue_credit(self, order_number, basket, reference_number, amount, currency)
...
-        return validated_receipt.get('receipt', {}).get('in_app', [{}])[0].get(attribute)
+        if self.NAME == 'ios-iap':
+            return validated_receipt.get('receipt', {}).get('in_app', [{}])[0].get(attribute)
+        elif self.NAME == 'android-iap':
+            validated_receipt.get('raw_response', {}).get(attribute)
+
+    def _get_transaction_id_from_receipt(self, validated_receipt):
+        if self.NAME == 'ios-iap':
+            return self._get_attribute_from_receipt(validated_receipt, 'transaction_id')
+        elif self.NAME == 'android-iap':
```
Comment (jawad-khan) [inline ecommerce/extensions/iap/processors/base_iap.py]: what if we get receipt from _get_attribute_from_receipt for android?

Comment (moeez96) [reply ecommerce/extensions/iap/processors/base_iap.py]: We fetch transaction_id and original_transaction_id in case of iOS as well from receipt.

**Pull request merged**